src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionViewModel.cs

```diff
@@ -225,14 +225,10 @@ internal sealed class ConnectionViewModel : BindableBase, IConnectionViewModel {
             }
 
             Uri uri;
-            var isPathValid = Uri.TryCreate(Path, UriKind.Absolute, out uri) && !string.IsNullOrEmpty(uri?.Host);
-            if (!IsValidConnectionName(Name)) {
-                IsValid = false;
-                SaveButtonTooltip = Resources.ConnectionManager_ShouldHaveName;
-            } else if (!isPathValid) {
-                IsValid = false;
-                SaveButtonTooltip = uri == null 
-                    ? Resources.ConnectionManager_ShouldHavePath 
+            IsPathValid = IsValidConnectionUrl(Path, out uri);
```

Comment:
Up to you but I'd keep the method in case we want to add more checks. IMO it is more readable.